### DISCUSSION TITLE
Collapsible element support

### DISCUSSION THREAD

**Discussion by greg-hutchinson** [hakimel/reveal.js#3345] [Ideas]
I am working a lot with Asciidoc and of course it was natural to try some of the Asciidoc to RevealJs for presentations. However, a feature that I haven't been able to find in RevealJs is the ability to have a Collapsible element like in Asciidoc. https://docs.asciidoctor.org/asciidoc/latest/blocks/collapsible/

Similar things can be done with a fragment, but sometimes I don't want the collapsible element to be shown at all - for example if I want to skip forward to a slide for an instant and then back again. I only want the element to be shown when I click on the element itself.

**Comment by Martinomagnifico** [comment]
Hi Greg, 

If you mean support in general for this kind of element, then any of the current browsers (and older as well) support the `<details>` tag. You just have to write in HTML and then it works:

```
<details>
  <summary>Reveal.js</summary>
  <p>reveal.js is an open source HTML presentation framework. It's a tool that enables anyone with a web browser to create fully-featured and beautiful presentations for free.</p>
</details>
```

But if you mean support for this kind of element in Markdown (similar to how Asciidoc uses custom tags that also render to HTML), then there is none. Reveal.js uses Marked as interpreter, and both Markdown and Marked have no syntax that converts to `<details>`. You can always just write HTML inside your Markdown files.

Styling for the `<details>` tag can be easily done in CSS.

**Comment by greg-hutchinson** [comment]
Thanks Martin,
Yes - that is what I was interested in.
I also did a little more digging in the Asciidoctor-revealjs converter.
Seems this is a known issue.
I will use the passthru option in the asciidoc and use the HTML as you
suggested.
Thanks again.


On Thu, 5 Jan 2023 at 07:04, Martino ***@***.***> wrote:

> Hi Greg,
>
> If you mean support in general for this kind of element, then any of the
> current browsers (and older as well) support the <details> tag. You just
> have to write in HTML and then it works:
>
> <details>
>   <summary>Reveal.js</summary>
>   <p>reveal.js is an open source HTML presentation framework. It's a tool that enables anyone with a web browser to create fully-featured and beautiful presentations for free.</p>
> </details>
>
> But if you mean support for this kind of element in Markdown (similar to
> how Asciidoc uses custom tags that also render to HTML), then there is
> none. Reveal.js uses Marked as interpreter, and both Markdown and Marked
> have no syntax that converts to <details>. You can always just write HTML
> inside your Markdown files.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hakimel/reveal.js/discussions/3345#discussioncomment-4601248>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADDG3JSYZPJCXRF4JNXIGLLWQ3BFBANCNFSM6AAAAAATRHR3QI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Greg Hutchinson - Agile Coach
